packages/components/dialog/src/dialog.ts

```diff
@@ -11,7 +11,9 @@ export const dialogProps = buildProps({
     default: false,
   },
   beforeClose: {
-    type: definePropType<(...args: any[]) => void>(Function),
+    type: definePropType<(done: (shouldCancel: boolean) => void) => void>(
```

Comment:
**Nitpick:** can we extract this into a type so that we can have a nicer format here?
e.g.

```ts
type DoneFn = (cancel: boolean) => void
type BeforeCloseFn = (done: DoneFn) => void
```

Reply:
Updated.